### PR TITLE
CRUD for Pet and User (not fully completed)

### PR DESCRIPTION
**Pull request by adinaned**

### CODE REVIEW

File: src/DucksNet.WebUI/Pages/Models/CreatePetModel.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace DucksNet.WebUI.Pages.Models;
+
+public class CreatePetModel
+{
+    public string? PetId { get; set; }
```
Comment (OpariucRares) [inline src/DucksNet.WebUI/Pages/Models/CreatePetModel.cs]: you do not need this

File: src/DucksNet.WebUI/Pages/Models/CreateUserModel.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace DucksNet.WebUI.Pages.Models;
+
+public class CreateUserModel
+{
+    public string? UserId { get; set; }
+
```
Comment (OpariucRares) [inline src/DucksNet.WebUI/Pages/Models/CreateUserModel.cs]: same here

**Review by OpariucRares — CHANGES_REQUESTED** [review]
look at comments

File: src/DucksNet.WebUI/Pages/Models/UpdatePetModel.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace DucksNet.WebUI.Pages.Models;
+
+public class UpdatePetModel
+{
+    [Required]
+    [StringLength(36, MinimumLength = 36, ErrorMessage = "Pet id must be 36 characters long.")]
+    public string? PetId { get; set; }
+    public string? Name { get; set; }
```
Comment (Al-Andrew) [inline src/DucksNet.WebUI/Pages/Models/UpdatePetModel.cs:10]: What happens If I pass null into any of the ones not marked required?

File: src/DucksNet.WebUI/Pages/Models/UpdateUserModel.cs [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+{
+    [Required]
+    [StringLength(36, MinimumLength = 36, ErrorMessage = "User id must be 36 characters long.")]
+    public string? UserId { get; set; }
+    public string? FirstName { get; set; }
+    public string? LastName { get; set; }
+    public string? Address { get; set; }
+    public string? PhoneNumber { get; set; }
+    public string? Email { get; set; }
+    public string? Password { get; set; }
```
Comment (Al-Andrew) [inline src/DucksNet.WebUI/Pages/Models/UpdateUserModel.cs:15]: What happens if I pass null into any of the ones not marked Required?

File: src/DucksNet.WebUI/Pages/Services/UserDataServices.cs [hunk 5]
```diff
@@ -12,17 +14,31 @@ public UserDataService(HttpClient httpClient)
...
     {
-#pragma warning disable CS8603 // Possible null reference return.
+        var users = await httpClient.GetFromJsonAsync<IEnumerable<User>>(ApiURL);
+        return users!;
+    }
+
+    public Task<User> GetUserDetail(Guid userId)
+    {
+        throw new NotImplementedException();
+    }
```
Comment (Al-Andrew) [inline src/DucksNet.WebUI/Pages/Services/UserDataServices.cs]: Is this function needed/used?

**Review by Al-Andrew — APPROVED** [review]
Seems good enough

**Review by OpariucRares — APPROVED** [review]
good

**Pull request merged**